### PR TITLE
Fix MMR peak bagging, add tests cases

### PR DESCRIPTION
**Pull request by philipstanislaus**
- Changes peak bagging to expect all right peaks to be bagged into one hash in the proof
- Copies @vgeddes' hardhat config to enable logging from Solidity (thanks 🎉 ) 
- MMR verification works now with the same data that substrate uses in https://github.com/paritytech/substrate/blob/87e63fa7c70be29383cb95ff52d71874f1db908a/frame/merkle-mountain-range/src/tests.rs#L263 
- Adds further test fixtures generated with substrate code

### CODE REVIEW

**Comment by musnit** [comment]
@philipstanislaus is this based onto the right branch? or should it be onto main?

File: ethereum/contracts/MMRVerification.sol [hunk 1]
```diff
@@ -92,17 +93,7 @@ contract MMRVerification {
         // TODO: Make this more elegant
         bytes32 rightPeaksHash;
         if (numRightPeaks > 0) {
-            if (numRightPeaks == 1) {
```
Comment (musnit) [inline ethereum/contracts/MMRVerification.sol:95]: so the only real logic change is removing these 2 cases and just doing a simple one-line bag for both cases?

Comment (philipstanislaus) [reply ethereum/contracts/MMRVerification.sol:95]: Exactly

**Comment by philipstanislaus** [comment]
It is based on the correct branch since @denalimarsh has made further changes to fix MMR bagging.

**Comment by philipstanislaus** [comment]
This is ready to be merged @musnit

**Comment by musnit** [comment]
@philipstanislaus ok - can we get a PR targeting main? either this one or merge into dm-update-peak-bagging and then PR that?

**Comment by philipstanislaus** [comment]
Replaced by #418

**Pull request closed without merging**